core/src/main/java/org/testcontainers/DockerClientFactory.java

```diff
@@ -98,11 +98,8 @@ public DockerClient client() {
 
             checkVersion(version.getVersion());
 
-            List<Image> images = client.listImagesCmd().exec();
             // Pull the image we use to perform some checks
-            if (images.stream().noneMatch(it -> it.getRepoTags() != null && asList(it.getRepoTags()).contains(TINY_IMAGE))) {
-                client.pullImageCmd(TINY_IMAGE).exec(new PullImageResultCallback()).awaitSuccess();
-            }
+            checkAndPullImage(client, TINY_IMAGE);
```

Comment:
`checkDiskSpaceAndHandleExceptions` calls `checkDiskSpace` which calls `runInsideDocker` which calls `checkAndPullImage` after your change.

It means that we could remove `checkAndPullImage` from here and inline `checkAndPullImage` in `runInsideDocker`.

WDYT?